pic-sure-hpds-ui/src/main/webapp/picsureui/analytics/cookie-consent.js

```diff
@@ -0,0 +1,54 @@
+define([
+    'backbone',
+    'handlebars',
+    'text!analytics/cookie-consent.hbs',
+    "picSure/settings",
+], function(BB, HBS, viewTemplate, settings) {
+    const CookieBanner = BB.View.extend({
+        initialize: function(opts){
+            this.template = HBS.compile(viewTemplate);
+        },
+        events: {
+            'click #cookie-consent-accept-all' : 'acceptAll',
+            'click #cookie-consent-reject-all' : 'rejectAll',
+        },
+        acceptAll: function(){
+            this.setConsent({
+                necessary: true,
+                preferences: true,
+                analytics: true,
+            });
+            this.hideBanner();
+        },
+        rejectAll: function(){
+            this.setConsent({
+                necessary: false,
+                preferences: false,
+                analytics: false,
+                marketing: false,
+            });
+            this.hideBanner();
+        },
+        setConsent(consent) {
+            const consentMode = {
+              'functionality_storage': consent.necessary ? 'granted' : 'denied',
```

Comment:
Are the options independent from each other? Can we not add an option to select some individually?

Reply:
Decided against it.